src/confirm.rs

```diff
@@ -68,6 +68,26 @@ async fn transaction_receipt_block_number_check<T: Transport>(eth: &Eth<T>, hash
     Ok(receipt.and_then(|receipt| receipt.block_number))
 }
 
+async fn decode_revert_reason(revert_reason_abi: &str) -> error::Result<String> {
+    let method_id_len = 10;
+    let block_len = 64;
+
+    // remove method id and data offset
+    let cleaned_abi = &revert_reason_abi[method_id_len + block_len..];
+
+    let len_hex = &cleaned_abi[..block_len];
+    let reason_len = 2 * usize::from_str_radix(len_hex, 16)
+        .map_err(|err| Error::Decoder(format!("{:?}", err)))?;
```

Comment:
pelase add clarification into error that we are unable to parse transaction revert reason

Reply:
added